spyder/widgets/ipythonconsole/client.py

```diff
@@ -323,33 +326,62 @@ def restart_kernel(self):
         Took this code from the qtconsole project
         Licensed under the BSD license
         """
-        message = _('Are you sure you want to restart the kernel?')
-        buttons = QMessageBox.Yes | QMessageBox.No
-        result = QMessageBox.question(self, _('Restart kernel?'),
-                                      message, buttons)
-        if result == QMessageBox.Yes:
-            sw = self.shellwidget
+        sw = self.shellwidget
+        # The lines below are needed to restart the kernel without a prompt
+        # when an error in stdout corrupts the debugging process.
+        # See issue 4003
+        if sw._input_none:
             if sw.kernel_manager:
-                if self.infowidget.isVisible():
-                    self.infowidget.hide()
-                    sw.show()
-                try:
-                    sw.kernel_manager.restart_kernel()
-                except RuntimeError as e:
+                    if self.infowidget.isVisible():
+                        self.infowidget.hide()
+                        sw.show()
+                    try:
+                        sw.kernel_manager.restart_kernel()
+                    except RuntimeError as e:
+                        sw._append_plain_text(
+                            _('Error restarting kernel: %s\n') % e,
+                            before_prompt=True
+                        )
+                    else:
+                        sw.reset(clear=False)
+                        sw._append_html(_("<br>Kernel restarting because an "
+                                          "error raised while debugging "
+                                          "(stdout change)\n<hr><br>"),
+                                        before_prompt=False)

```

Comment:
Please make this change again because it went away after I removed the test commits. Sorry.